share/test_helpers.go

```diff
@@ -44,9 +43,10 @@ func RandEDS(t *testing.T, size int) *rsmt2d.ExtendedDataSquare {
 }
 
 // RandShares generate 'total' amount of shares filled with random data.
-func RandShares(t *testing.T, total int) []Share {
+func RandShares(t require.TestingT, total int) []Share {
 	if total&(total-1) != 0 {
-		t.Fatal("Namespace total must be power of 2")
+		t.Errorf("Namespace total must be power of 2")
+		t.FailNow()
```

Comment:
Why change from `Fatal` to `Error` and `FailNow`? Seems like an unnecessary line add. 

Did you mean this? 
```suggestion
		t.Fatal("Namespace total must be power of 2:", total)
```

Reply:
Yes, it is necessary because of testing.T

Reply:
[non-blocking]
Personal preference, I would structure helper functions as stand alone functions. I don't think they need to have the tester passed through. I think it is cleaner to just return an error and have the calling test handle the testing failure.  

This then also allows us to test these functions to ensure that our tests are using sound functions and we aren't building tests off of faulty helpers.

Reply:
#1275